### ISSUE TITLE
Godot4: applying generated terrain has no effect

### ISSUE THREAD

**Issue by ian-h-chamberlain** [Zylann/godot_heightmap_plugin#371]
**Describe the bug**
Hi, trying to use this plugin for the first time and I can't seem to get the generated terrain to take effect in the 3D editor. Painting terrain works, but I can't seem to get the generated changes to stick.

Opening a bug as requested in https://github.com/Zylann/godot_heightmap_plugin/issues/316#issuecomment-1483936939

**To Reproduce**
1. Create an HTerrain
2. Select a data directory for the terrain
3. Open `Terrain > Generate...` dialog
4. Click Apply

I see this in the output panel:
```
  core/io/image.cpp:2775 - Condition "format != p_src->format" is true.
  core/io/image.cpp:2775 - Condition "format != p_src->format" is true.
Generate terrain
```

I also tried running `godot -d` and got slightly more info which might help debug:
```
ERROR: Condition "format != p_src->format" is true.
   at: blit_rect (core/io/image.cpp:2775)
ERROR: Condition "format != p_src->format" is true.
   at: blit_rect (core/io/image.cpp:2775)
```

**Expected behavior**
Clicking the Apply button actually modifies the terrain in the scene editor.

**Screenshots**
Not much to show, since there's no terrain, but I can record a screen capture if that's helpful.

**Environment**
- OS: `Linux steamdeck 5.13.0-valve36-1-neptune #1 SMP PREEMPT Mon, 19 Dec 2022 23:39:41 +0000 x86_64 GNU/Linux`
- `OpenGL API 4.6 (Core Profile) Mesa 23.0.2 (git-4d5e73870e) - Compatibility - Using Device: AMD - AMD Custom GPU 0405 (vangogh, LLVM 15.0.7, DRM 3.45, 5.13.0-valve36-1-neptune)`
- Godot `v4.0.2.stable.flathub [7a0977ce2]`
- Plugin version baeaba83345e1a5017587dd235749df5a528ef66 (`godot4` branch)
- Renderer used: not 100% sure - I'm using `gl_compatibility` in Godot and didn't change any default settings in the addon, so I guess OpenGL?

**Comment by Zylann** [comment]
Are you able to reproduce this with the Vulkan renderer? (the non-"compatibility" one?)

**Comment by Zylann** [comment]
Apparently, in Forward+, the SubViewport created by the generator provides texture images as RGB8, but for some reason in Compatibility it returns RGBA8... so the conversion routine has to be added back to make sure it ends up as RGB8. I don't know if that's expected or a Godot bug.

**Comment by Zylann** [comment]
Should be fixed in 5273707cab9d179dcae97000c59cab6f6645744e

**Comment by ian-h-chamberlain** [comment]
@Zylann thanks! Confirmed I can apply the terrain now after using the latest version of the `godot4` branch and Compatibility renderer. I think this issue can be closed. Cheers!
